common/src/main/java/net/william278/huskhomes/network/Broker.java

```diff
@@ -113,10 +116,58 @@ protected void handle(@NotNull OnlineUser receiver, @NotNull Message message) {
                 plugin.getManager().homes().updatePublicHomeCache();
                 plugin.getManager().warps().updateWarpCache();
             }
+            case RTP_LOCATION -> message.getPayload()
+                    .getRTPResponse()
+                    .ifPresentOrElse(response -> {
+                        final TeleportBuilder builder = Teleport.builder(plugin)
+                                .teleporter(receiver)
+                                .actions(TransactionResolver.Action.RANDOM_TELEPORT)
+                                .target(response.getPosition());
+                        builder.buildAndComplete(true);
+                    }, () -> plugin.getLocales().getLocale("error_rtp_randomization_timeout")
+                            .ifPresent(receiver::sendMessage));
             default -> throw new IllegalStateException("Unexpected value: " + message.getType());
         }
     }
 
+    /**
+     * Separate handler for RTP Request because it doesn't need a receiver to handle it.
+     *
+     * @param message the message to handle
+     */
+    protected void handle(@NotNull Message message) {
+        if (message.getSourceServer().equals(getServer())) {
+            return;
+        }
+        if (message.getType() == Message.Type.REQUEST_RTP_LOCATION) {
```

Comment:
Not a massive fan of checking the message type twice. Move these checks to before the method call and rename this to `handleRtpRequest` or something

Reply:
The reason i check twice is because of future development, I don't want other people to come into the code and just assume its fine to use.